src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -111,11 +119,13 @@ private void RegisterObjectBrowserLibraryManager()
             }
         }
 
-        private void UnregisterObjectBrowserLibraryManager()
+        private async Task UnregisterObjectBrowserLibraryManagerAsync(CancellationToken cancellationToken)
         {
             if (_libraryManagerCookie != 0)
```

Comment:
so _librarymanagerCookie is not guarded by lock or memory barrier but used in 2 different threads. considering not even using double lock pattern, shouldn't this be either volatile or under lock?